Tests/TestExecution.gd

```diff
@@ -11,30 +26,240 @@ func before_all():
+	DirAccess.remove_absolute(PROJECTS[1] + "EmptyDir/DirCopyNotRecursive/DirFile2.txt")
+	DirAccess.remove_absolute(PROJECTS[1] + "EmptyDir/DirCopyNotRecursive/SubDir/SubDirFile1.txt")
+	DirAccess.remove_absolute(PROJECTS[1] + "EmptyDir/DirCopyNotRecursive/SubDir")
+	DirAccess.remove_absolute(PROJECTS[1] + "EmptyDir/DirCopyNotRecursive")
+
+func test_clear_directory_files():
+	# Check assumptions
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "MessyDir/DirFile1.txt"))
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "MessyDir/DirFile2.txt"))
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "MessyDir/SubDir/SubDirFile1.txt"))
+	
+	# Setup
+	DirAccess.copy_absolute(PROJECTS[1] + "MessyDir/DirFile1.txt", PROJECTS[1] + "EmptyDir/DirFile1.txt")
+	DirAccess.copy_absolute(PROJECTS[1] + "MessyDir/DirFile2.txt", PROJECTS[1] + "EmptyDir/DirFile2.txt")
+	Data.load_project(PROJECTS[1])
+	Data.current_routine = Data.routines[ROUTINES.clear_directory_files]
+	
+	# Execute
+	add_child.call_deferred(scene)
+	await wait_for_signal(scene.finished, EXECUTION_TIMEOUT)
+	
+	# Check results
+	assert_false(FileAccess.file_exists(PROJECTS[1] + "MessyDir/DirFile1.txt"))
+	assert_false(FileAccess.file_exists(PROJECTS[1] + "MessyDir/DirFile2.txt"))
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "MessyDir/SubDir/SubDirFile1.txt"))
+
+	# Cleanup
+	DirAccess.rename_absolute(PROJECTS[1] + "EmptyDir/DirFile1.txt", PROJECTS[1] + "MessyDir/DirFile1.txt")
+	DirAccess.rename_absolute(PROJECTS[1] + "EmptyDir/DirFile2.txt", PROJECTS[1] + "MessyDir/DirFile2.txt")
+
+func test_pack_zip():
+	# Check assumptions
+	assert_true(DirAccess.dir_exists_absolute(PROJECTS[1] + "MessyDir"))
+	assert_false(FileAccess.file_exists(PROJECTS[1] + "MessyDir.zip"))
+	assert_false(FileAccess.file_exists(PROJECTS[1] + "Include.zip"))
+	assert_false(FileAccess.file_exists(PROJECTS[1] + "Exclude.zip"))
+	
+	# Setup
+	Data.load_project(PROJECTS[1])
+	Data.current_routine = Data.routines[ROUTINES.pack_zip]
+	
+	# Execute
+	add_child.call_deferred(scene)
+	await wait_for_signal(scene.finished, EXECUTION_TIMEOUT)
+	
+	# Check results
+	assert_true(DirAccess.dir_exists_absolute(PROJECTS[1] + "MessyDir"))
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "MessyDir.zip"))
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "Include.zip"))
+	assert_true(FileAccess.file_exists(PROJECTS[1] + "Exclude.zip"))
+	
+	var reader := ZIPReader.new()
+	var error := reader.open(PROJECTS[1] + "MessyDir.zip")
+	var files := reader.get_files()
+	reader.close()
+	assert_true(error == OK)
+	assert_true(files.has("DirFile1.txt"))
+	assert_true(files.has("DirFile2.txt"))
+	assert_true(files.has("SubDir/SubDirFile1.txt"))
+	
+	reader = ZIPReader.new()
+	error = reader.open(PROJECTS[1] + "Include.zip")
+	files = reader.get_files()
+	reader.close()
+	assert_true(error == OK)
+	assert_true(files.has("DirFile1.txt"))
+	assert_false(files.has("DirFile2.txt"))
+	assert_false(files.has("SubDir/SubDirFile1.txt"))
```

Comment:
ZIP packing also allows for globbing (e.g. `*.txt`). Would be nice to test this too. (can be added later though)